### PR TITLE
#73: Use node run state for credentials

### PR DESCRIPTION
**Pull request by llibicpep**
## Pull Request Checklist

https://github.com/SumoLogic/sumologic-collector-chef-cookbook/issues/73

#### General

- [x] Remove any versioning you did yourself if applicable

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/) with all new changes under `## [Unreleased]` and using a `### Added, Fixed, Changed, or Breaking Change` sub-header.

- [x] Update README with any necessary changes

- [x] RuboCop passes

- [x] Foodcritic passes

- [x] Existing tests pass

#### Purpose

To enable wrapper cookbooks to maintain their own standards as to how they store and manage secrets while not making the code vulnerable by using attributes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,8 +43,8 @@ Installation
   knife cookbook github install SumoLogic/sumologic-collector-chef-cookbook
   ```
 3. Specify data bag and item with your access credentials.  The data item should
-contain attributes `accessID` and `accessKey`.  Note that attribute names are case sensitive.  If the cases mismatch, the values will not appear when chef-client runs.  The default data bag/item is
-`['sumo-creds']['api-creds']`
+contain attributes `accessID` and `accessKey`. Note that attribute names are case sensitive.  If the cases mismatch, the values will not appear when chef-client runs. The default data bag/item is
+`['sumo-creds']['api-creds']`. More flexible option is to use `node.run_state['sumo_key_id']` and `node.run_state['sumo_key_secret']` to supply credentials.
```
Comment (majormoses) [inline README.md]: A more flexible but less secure option is...

Comment (llibicpep) [reply README.md]: There are indeed very good reasons, but besides encrypted data bags there are plenty of other secure ways to do it, using Vault, Hashicorp Vault, or other secret server analogue. And this is true that sensitive data should not be stored in the node object, and the reason for that is the node object being uploaded to the Chef server in plain text. But `node.run_state` has been designed for very same reason, it is not persistent and being discarded at the end of chef-client run, it's not communicated outside of the node. More on this https://docs.chef.io/recipes.html#node-run-state.

Comment (majormoses) [reply README.md]: Fair enough, If you want to change the verbiage to say any method that does not provide encrypted content is insecure would be fine with me. I just want it very clear that I don't recommend any method that does not involve encryption for something as sensitive as access keys.

Comment (llibicpep) [reply README.md]: Makes sense - I will add a note on how to use `run_state` correctly. Also would you like to add similar warning Jenkins cookbook has when someone tries to use attributes?

Comment (majormoses) [reply README.md]: sgtm

Comment (llibicpep) [reply README.md]: done

**Comment by majormoses** [comment]
I have released: https://github.com/SumoLogic/sumologic-collector-chef-cookbook/releases/tag/v1.3.0

@duchatran ping on release to supermarket

**Pull request merged**